src/widgets.js

```diff
@@ -591,11 +728,51 @@ define([
         return $el;
     }
 
+    function replaceOuputRef(form, value, withClose, noOutput) {
+        var template = "<output value=\"" + value + "\" />";
+        if (noOutput) {
+            template = value;
+        }
+        var v = value.split('/'),
+            dispValue = v[v.length-1],
+            mug = form.getMugByPath(value),
+            icon = mug ? mug.options.icon: 'fcc fcc-fd-external-case',
+            datanodeClass = mug ? 'label-datanode-internal' : 'label-datanode-external',
+            richText = $('<span>').addClass('label label-datanode')
+                .addClass(datanodeClass)
+                .attr({
+                    contenteditable: false,
+                    draggable: true,
+                    'data-value': template
+                }).append($('<i>').addClass(icon).html('&nbsp;')).append(dispValue);
+        if (withClose) {
+            richText.append($("<button>").addClass('close').html("&times;"));
+        }
+        return richText;
+    }
+
+    function toRichText(val, form, withClose) {
+        if (!val) {return "";}
+        val = val.replace('&lt;', '<').replace('&gt;', '>');
```

Comment:
It's hard for me to tell all the things that are happening here and below, but it looks similar to the [XML normalization utilities](https://github.com/dimagi/Vellum/blob/031ab9af76f987fe002c80d0d5bbc29f6ca0d369/src/xml.js#L11-L64). Would one or both of those be useful here?